python/tvm/meta_schedule/testing/torchbench/run.py

```diff
@@ -0,0 +1,508 @@
+
+        jit_mod = torch.jit.trace(graph_module, example_inputs)
+        shape_list = [(f"inp_{idx}", i.shape) for idx, i in enumerate(example_inputs)]
+        ir_mod, params = tvm.relay.frontend.from_pytorch(jit_mod, shape_list)
+
+        relay_build = {"graph": relay.build, "vm": relay.vm.compile}[ARGS.backend]
+        with ARGS.target, ms.utils.autotvm_silencer(), database:
+            with PassContext(
+                opt_level=3,
+                config={
+                    "relay.backend.use_meta_schedule": True,
+                    "relay.backend.use_meta_schedule_dispatch": not IS_CUDA,
+                    "relay.backend.tir_converter": "default",
+                },
+            ):
+                lib = relay_build(ir_mod, target=ARGS.target, params=params)
+
+        device = tvm.cuda(0) if IS_CUDA else tvm.cpu(0)
+
+        if ARGS.backend == "graph":
+            mod = graph_executor.GraphModule(lib["default"](device))
+            return get_graph_executor_forward(mod)
+        elif ARGS.backend == "vm":
+            vm = VirtualMachine(lib, device)
+            return get_vm_forward(vm)
+        else:
+            raise RuntimeError(f"Unknown backend {ARGS.backend}")
+
+    return backend
+
+
+def format_time(seconds):
+    if seconds > 1:
+        return f"{seconds:.3g}s"
+    else:
+        return f"{seconds * 1000:.3g}ms"
+
+
+def null_experiment(*args, **kwargs):
+    return ""
+
+
+def performance_experiment(model_iter_fn, model, example_inputs):
+    # Simplified from https://github.com/pytorch/torchdynamo/blob/c537639f9712621dc04ca09908796dbbe86c354b/benchmarks/common.py#L494
+    timings = np.zeros((ARGS.benchmark_repeat, 2), np.float64)
+
+    is_correct = True
+
+    frozen_model_iter_fn = torchdynamo.run(model_iter_fn)
+    for rep in range(ARGS.benchmark_repeat):
+        # interleave the runs to handle frequency scaling and load changes
+        timings[rep, 0], expected_output = timed(
+            model, model_iter_fn, example_inputs, return_result=True
+        )
+        timings[rep, 1], actual_output = timed(
+            model, frozen_model_iter_fn, example_inputs, return_result=True
+        )
+        is_correct = is_correct and same(
+            expected_output, actual_output, cos_similarity=ARGS.cosine_similarity
+        )
+
+    pvalue = ttest_ind(timings[:, 0], timings[:, 1]).pvalue
+    median = np.median(timings, axis=0)
+    speedup = median[0] / median[1]
+    logger.info(
+        f"eager:{format_time(median[0])} optimized:{format_time(median[1])} speedup:{speedup:.3f}x p:{pvalue:.3f}"
+    )
+    if not is_correct:
+        logger.error(f"Result is incorrect.")
+        logger.error(f"Expected (PyTorch eager): {expected_output}")
+        logger.error(f"Actual (Optimized): {actual_output}")
+
+    return ""
+
+
+def get_torch_device_type(target: tvm.target.Target) -> str:
+    if target.kind.name == "llvm":
+        return "cpu"
+    else:
+        return target.kind.name
```

Comment:
Since we may only test on cuda / llvm, maybe it's better to throw an exception when the given target is not these devices, e.g., hexagon?